tsconfig.json

```diff
@@ -1,110 +1,12 @@
-    "forceConsistentCasingInFileNames": true,            /* Ensure that casing is correct in imports. */
-
-    /* Type Checking */
-    "strict": true,                                      /* Enable all strict type-checking options. */
-    "noImplicitAny": false,                            /* Enable error reporting for expressions and declarations with an implied `any` type.. */
-    "strictNullChecks": true,                         /* When type checking, take into account `null` and `undefined`. */
-    "strictFunctionTypes": true,                      /* When assigning functions, check to ensure parameters and the return values are subtype-compatible. */
-    "strictBindCallApply": true,                      /* Check that the arguments for `bind`, `call`, and `apply` methods match the original function. */
-    "strictPropertyInitialization": false,             /* TODO: do we want this active? Check for class properties that are declared but not set in the constructor. */
-    "noImplicitThis": true,                           /* Enable error reporting when `this` is given the type `any`. */
-    "useUnknownInCatchVariables": true,               /* Type catch clause variables as 'unknown' instead of 'any'. */
-    // "alwaysStrict": true,                             /* Ensure 'use strict' is always emitted. */
-    "noUnusedLocals": true,                           /* Enable error reporting when a local variables aren't read. */
-    "noUnusedParameters": true,                       /* Raise an error when a function parameter isn't read */
-    // "exactOptionalPropertyTypes": true,               /* Interpret optional property types as written, rather than adding 'undefined'. */
-    "noImplicitReturns": true,                        /* Enable error reporting for codepaths that do not explicitly return in a function. */
-    "noFallthroughCasesInSwitch": true,               /* Enable error reporting for fallthrough cases in switch statements. */
-    // "noUncheckedIndexedAccess": true,                 /* Include 'undefined' in index signature results */
-    "noImplicitOverride": true,                       /* Ensure overriding members in derived classes are marked with an override modifier. */
-    // "noPropertyAccessFromIndexSignature": true,       /* Enforces using indexed accessors for keys declared using an indexed type */
-    "allowUnusedLabels": false,                        /* Disable error reporting for unused labels. */
-    "allowUnreachableCode": false,                     /* Disable error reporting for unreachable code. */
-
-    /* Completeness */
-    // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
-    "skipLibCheck": true                                 /* Skip type checking all .d.ts files. */
-  },
-  "include": [
-    "./src/**/*.ts"
-  ],
-  "exclude": [
-    "./src/**/*.spec.ts"
-  ]
+    "compilerOptions": {
```

Comment:
I find these comments descriptive, and it shows what we are still missing in order to be fully strict
I would actually prefer to restore the changes on this file
also target changed to 2019? Any reason for that, are you using older browser?

Reply:
2019 is the last version that webpack 4.0 support - I know this is bit lame; But we have a complex build pipeline on one of my projects, and we don't want the overhead of transpiling node_modules to support es2019+

If you are happy for the downgrade - that would be handy for my use-case.

RE: The tsconfig purge.
When converting to the examples - I couldn't figure our what was significant and what wasn't. I went back to the default (basically empty) then built it back up....

There is a bit more effort to get the project fully strict ... i did try to fix a few extras too.

Reply:
I think it's not so scary to slightly improve compatibility and change the target to 2019. And at the expense of strict mode - I agree with @Smrtnyk. Perhaps we should leave only those options that do not allow us to enable strict mode?